src/main/java/domain/AttendanceBook.java

```diff
@@ -0,0 +1,90 @@
+package domain;
+
+import exception.DuplicateAttendanceException;
+import exception.InvalidDateException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.*;
+
+public class AttendanceBook {
+    private final HashSet<Attendance> attendances;
+
+    public AttendanceBook() {
+        attendances = new HashSet<>();
+    }
+
+    public Attendance register(LocalDate date, LocalTime time) {
+        final boolean isExist = attendances.stream().anyMatch(attendance -> attendance.isAttendedOn(date));
+        if (isExist) {
+            throw new DuplicateAttendanceException();
+        }
+        if (isHoliday(date)) {
+            throw new InvalidDateException();
+        }
+        Attendance attendance = Attendance.of(date, time);
+        attendances.add(attendance);
+        return attendance;
+    }
+
+    public Attendance replace(LocalDate date, LocalTime time) {
+        Attendance oldAttendance = findAttendanceByDate(date);
+        Attendance modifiedAttendance = oldAttendance.modify(time);
+        attendances.remove(oldAttendance);
+        attendances.add(modifiedAttendance);
+        return modifiedAttendance;
+    }
+
+    public Attendance findAttendanceByDate(LocalDate date) {
+        return attendances.stream()
+                .filter(attendance -> attendance.isAttendedOn(date))
+                .findFirst()
+                .orElse(Attendance.empty(date));
+    }
+
+    public List<Attendance> getAllAttendances(LocalDate startDate, LocalDate endDate) {
+        List<Attendance> result = new ArrayList<>();
+        for (LocalDate current = startDate; current.isBefore(endDate); current = current.plusDays(1)) {
+            if (isHoliday(current)) {
+                continue;
+            }
+            Attendance attendance = findAttendanceByDate(current);
+            result.add(attendance);
+        }
+        return result;
```

Comment:
```suggestion
        return startDate.datesUntil(endDate)
                .filter(date -> !isHoliday(date))
                .map(this::findAttendanceByDate)
                .collect(Collectors.toList());
```

저도 이번에 처음 알았는데 `datesUntil()`을 이용해서 Stream<LocalDate>를 만들 수 있었네요!